test/index.ts

```diff
@@ -508,6 +632,154 @@ describe('Storage', () => {
     });
   });
 
+  describe('getHmacKeys', () => {
+    // tslint:disable-next-line: no-any
+    let storageRequestStub: sinon.SinonStub<any, any>;
+    const SERVICE_ACCOUNT_EMAIL = 'service-account@gserviceaccount.com';
+    const ACCESS_ID = 'some-access-id';
+    const metadataResponse = {
+      accessId: ACCESS_ID,
+      etag: 'etag',
+      id: ACCESS_ID,
+      projectId: 'project-id',
+      serviceAccountEmail: SERVICE_ACCOUNT_EMAIL,
+      state: 'ACTIVE',
+      timeCreated: '20190101T00:00:00Z',
+      updated: '20190101T00:00:00Z',
+    };
+
+    beforeEach(() => {
+      storageRequestStub = sinon.stub(storage, 'request');
+      storageRequestStub.callsFake((_opts: {}, callback: Function) => {
+        callback(null, {});
+      });
+    });
+
+    it('should get HmacKeys without a query', done => {
+      storage.getHmacKeys(() => {
+        const firstArg = storage.request.firstCall.args[0];
+        assert.strictEqual(
+          firstArg.uri,
+          `/projects/${storage.projectId}/hmacKeys`
+        );
+        assert.deepStrictEqual(firstArg.qs, {});
+        done();
+      });
+    });
+
+    it('should get HmacKeys with a query', done => {
+      const query = {
+        maxResults: 5,
+        pageToken: 'next-page-token',
+        serviceAccountEmail: SERVICE_ACCOUNT_EMAIL,
+        showDeletedKeys: false,
+      };
+
+      storage.getHmacKeys(query, () => {
+        const firstArg = storage.request.firstCall.args[0];
+        assert.strictEqual(
+          firstArg.uri,
+          `/projects/${storage.projectId}/hmacKeys`
+        );
+        assert.deepStrictEqual(firstArg.qs, query);
+        done();
+      });
+    });
+
+    it('should execute callback with error', done => {
+      const error = new Error('Error.');
+      const apiResponse = {};
+
+      storageRequestStub.callsFake((_opts: {}, callback: Function) => {
+        callback(error, apiResponse);
+      });
+
+      storage.getHmacKeys(
+        {},
+        (err: Error, hmacKeys: HmacKey[], nextQuery: {}, resp: Metadata) => {
+          assert.strictEqual(err, error);
+          assert.strictEqual(hmacKeys, null);
+          assert.strictEqual(nextQuery, null);
+          assert.strictEqual(resp, apiResponse);
+          done();
+        }
+      );
+    });
+
+    it('should return nextQuery if more results exist', done => {
+      const token = 'next-page-token';
+      storageRequestStub.callsFake((_opts: {}, callback: Function) => {
+        callback(null, {nextPageToken: token, items: []});
+      });
+
+      storage.getHmacKeys(
+        {maxResults: 5},
+        // tslint:disable-next-line: no-any
+        (err: Error, _hmacKeys: [], nextQuery: any) => {
+          assert.ifError(err);
+          assert.strictEqual(nextQuery.pageToken, token);
+          assert.strictEqual(nextQuery.maxResults, 5);
```

Comment:
Using the real `maxResults` option makes it seem like that's actually doing something, when I think the real test is that the user-provided options are still in tact inside of `nextQuery`. Could you separate the query object and confirm that the properties still exist?

```js
const query = {a: 'b', c: 'd'}
const expectedNextQuery = Object.assign({}, query, {pageToken: token})
storage.getHmacKeys(query, (err, _hmacKeys, nextQuery) => {
  assert.strictEqual(nextQuery, expectedNextQuery)
})
```